coap/adapter.go

```diff
@@ -1,61 +1,129 @@
-package adapter
+package coap
 
 import (
-	"encoding/json"
-	"log"
+	"errors"
+	"fmt"
 	"net"
 
-	"github.com/dustin/go-coap"
+	mux "github.com/dereulenspiegel/coap-mux"
+	gocoap "github.com/dustin/go-coap"
+	"github.com/go-kit/kit/log"
 	"github.com/mainflux/mainflux"
+	manager "github.com/mainflux/mainflux/manager/client"
 	broker "github.com/nats-io/go-nats"
-	"go.uber.org/zap"
 )
 
-const protocol string = "coap"
-
-type Observer struct {
-	conn    *net.UDPConn
-	addr    *net.UDPAddr
-	message *coap.Message
+// Service reprents actual CoAP adapter service.
+type Service interface {
+	Observe(l *net.UDPConn, a *net.UDPAddr, m *gocoap.Message) *gocoap.Message
+	Recieve(l *net.UDPConn, a *net.UDPAddr, m *gocoap.Message) *gocoap.Message
 }
 
-type CoAPAdapter struct {
-	obsMap map[string][]Observer
-	logger *zap.Logger
+const (
+	token    string = "token"
+	prefix   string = "src.coap."
```

Comment:
Please align to latest changes - NATS topics are now equivalent to Mainflux channels: https://github.com/mainflux/mainflux/pull/207#pullrequestreview-106327364